userspace/libsinsp/cri.cpp

```diff
@@ -0,0 +1,244 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "cri.h"
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace libsinsp {
+namespace cri {
+std::string s_cri_unix_socket_path = "/run/containerd/containerd.sock";
+std::unique_ptr <runtime::v1alpha2::RuntimeService::Stub> s_cri = nullptr;
+int64_t s_cri_timeout = 1000;
+sinsp_container_type s_cri_runtime_type = CT_CRI;
+
+sinsp_container_type get_cri_runtime_type(const std::string &runtime_name)
+{
+	if(runtime_name == "containerd")
+	{
+		return CT_CONTAINERD;
+	} else
+	{
+		return CT_CRI;
+	}
+}
+
+bool parse_cri_image(const runtime::v1alpha2::ContainerStatus &status, sinsp_container_info *container)
+{
+	// image_ref may be one of two forms:
+	// host/image@sha256:digest
+	// sha256:digest
+	const auto &image_ref = status.image_ref();
+	auto digest_start = image_ref.find("sha256:");
+	if(digest_start != string::npos)
+	{
+		container->m_imagedigest = image_ref.substr(digest_start);
+	}
+
+	string hostname, port, digest;
+	sinsp_utils::split_container_image(status.image().image(),
+					   hostname,
+					   port,
+					   container->m_imagerepo,
+					   container->m_imagetag,
+					   digest,
+					   false);
+	container->m_image = status.image().image();
+	return true;
+}
+
+bool parse_cri_mounts(const runtime::v1alpha2::ContainerStatus &status, sinsp_container_info *container)
+{
+	for(const auto &mount : status.mounts())
+	{
+		const char *propagation;
+		switch(mount.propagation())
+		{
+			case runtime::v1alpha2::MountPropagation::PROPAGATION_PRIVATE:
+				propagation = "private";
+				break;
+			case runtime::v1alpha2::MountPropagation::PROPAGATION_HOST_TO_CONTAINER:
+				propagation = "rslave";
+				break;
+			case runtime::v1alpha2::MountPropagation::PROPAGATION_BIDIRECTIONAL:
+				propagation = "rshared";
+				break;
+			default:
+				propagation = "unknown";
+				break;
+		}
+		container->m_mounts.emplace_back(
+			mount.host_path(),
+			mount.container_path(),
+			"",
+			!mount.readonly(),
+			propagation);
+	}
+	return true;
+
+}
+
+bool walk_down_json(const Json::Value &root, const Json::Value **out, const std::string &key)
+{
+	if(root.isMember(key))
+	{
+		*out = &root[key];
+		return true;
+	}
+	return false;
+}
+
+template<typename... Args>
+bool walk_down_json(const Json::Value &root, const Json::Value **out, const std::string &key, Args... args)
+{
+	if(root.isMember(key))
+	{
+		return walk_down_json(root[key], out, args...);
+	}
+	return false;
+}
+
+bool set_numeric(const Json::Value &dict, const std::string &key, int64_t &val)
+{
+	if(!dict.isMember(key))
+	{
+		return false;
+	}
+	const auto &json_val = dict[key];
+	if(!json_val.isNumeric())
+	{
+		return false;
+	}
+	val = json_val.asInt64();
+	return true;
+}
+
+bool parse_cri_env(const Json::Value &info, sinsp_container_info *container)
```

Comment:
the container env is not being used anywhere except when looking for SYSDIG_AGENT_CONF. may be just limit this function to find this specific env only?

Reply:
I'd rather do this as a separate PR that limits Docker environments as well, is it OK?